src/modules-load/modules-load.c

```diff
@@ -71,7 +90,27 @@ static int apply_file(struct kmod_ctx *ctx, const char *path, bool ignore_enoent
                 if (strchr(COMMENTS, *line))
                         continue;
 
-                k = module_load_and_warn(ctx, line, true);
+                RET_GATHER(r, send_module_to_load(sock, line));
+        }
+
+        return r;
+}
+
+static int run_prober(int sock) {
+        _cleanup_(sym_kmod_unrefp) struct kmod_ctx *ctx = NULL;
+        char buffer[NAME_MAX + 1];
+        ssize_t bytes_received;
+
+        int r = module_setup_context(&ctx);
```

Comment:
`int r;` is that special variable for us we *always* use for error handling. Hence it gets a bit of a special treatment and is always defined at outermost scope at the top of functions, separately from its first use. Hence please do `int r;` above the empty line above, and then just use it here.